WWW/2_linear_chain.wdl

```diff
@@ -84,72 +124,207 @@ task BwaMem {
   
   String base_file_name = basename(input_fastq, ".fastq")
   String ref_fasta_local = basename(ref_fasta)
+  String read_group_id = "ID:" + base_file_name
+  String sample_name = "SM:" + base_file_name
+  String platform = "illumina"
+  String platform_info = "PL:" + platform   # Create the platform information
+
 
-  # CL added some fake read groups for MarkDuplicates to run. 
   command <<<
     set -eo pipefail
 
-    mv ~{ref_fasta} .
-    mv ~{ref_fasta_index} .
-    mv ~{ref_dict} .
-    mv ~{ref_amb} .
-    mv ~{ref_ann} .
-    mv ~{ref_bwt} .
-    mv ~{ref_pac} .
-    mv ~{ref_sa} .
-
     bwa mem \
-      -p -v 3 -t 16 -M -R '@RG\tID:foo\tSM:foo2' \
-      ~{ref_fasta_local} ~{input_fastq} > ~{base_file_name}.sam 
+      -p -v 3 -t 16 -M -R '@RG\t~{read_group_id}\t~{sample_name}\t~{platform_info}' \
```

Comment:
Consider changing the thread usage so this can run on weaker hardware rather than failing immediately.

Reply:
Arent we assuming at this point that the intended user is going to be running this in HPC/some cloud platform? I am thinking A would be best. Good to show struct examples etc. We can chat more about this just so I understand this a bit better?

Reply:
This bwa mem call can run on a typical laptop if it uses less threads if we just change the -t argument and run on heavily downsampled data. I was thinking it might be a good idea to have at least the first task of the workflow be runnable on typical hardware as a hands-on example. 

Of course, that may not be useful or desirable -- the inline examples in the workflow can run on weak hardware, so toning down the main focus might not be desirable. It kind of ties into what we're writing about test data, though -- it's useful to have test data that can run quickly, and it's also useful to have a workflow that can run on a local machine if you are developing on a local machine.

This isn't a dealbreaker by any means, just worth considering in the context of running tests.